packages/lib/SyncTargetWebDAV.js

```diff
@@ -55,6 +56,12 @@ class SyncTargetWebDAV extends BaseSyncTarget {
 			errorMessage: '',
 		};
 
+		const providerCheck = checkProviderIsUnsupported(options.path());
+		if (providerCheck.isUnsupported === true) {
+			output.errorMessage = `The WebDAV implementation of ${providerCheck.unsupportedProvider} is incompatible with Joplin, and as such is no longer supported. Please use a different sync method.`;
+			return output;
+		}
```

Comment:
I feel like you could wrap all this logic in `checkProviderIsUnsupported()` and that function would throw an error if there's any problem. That will avoid the duplicated if statements and error string. You can move that in the try/catch block below since it already handles catching exceptions

Reply:
Done. I also changed the name to `checkProviderIsSupported()`.